drivers/adc/adc_cc13xx_cc26xx.c

```diff
@@ -0,0 +1,310 @@
+	{ 170, AUXADC_SAMPLE_TIME_170_US },
+	{ 341, AUXADC_SAMPLE_TIME_341_US },
+	{ 682, AUXADC_SAMPLE_TIME_682_US },
+	{ 1370, AUXADC_SAMPLE_TIME_1P37_MS },
+	{ 2730, AUXADC_SAMPLE_TIME_2P73_MS },
+	{ 5460, AUXADC_SAMPLE_TIME_5P46_MS },
+	{ 10900, AUXADC_SAMPLE_TIME_10P9_MS },
+};
+
+struct adc_cc13xx_cc26xx_data {
+	struct adc_context ctx;
+	const struct device *dev;
+	uint32_t ref_source;
+	uint8_t sample_time;
+	uint16_t *buffer;
+	uint16_t *repeat_buffer;
+};
+
+struct adc_cc13xx_cc26xx_cfg {
+	unsigned long base;
+	void (*irq_cfg_func)(void);
+};
+
+
+static void adc_cc13xx_cc26xx_isr(const struct device *dev);
+
+
+static void adc_context_start_sampling(struct adc_context *ctx)
+{
+	struct adc_cc13xx_cc26xx_data *data =
+		CONTAINER_OF(ctx, struct adc_cc13xx_cc26xx_data, ctx);
+
+	data->repeat_buffer = data->buffer;
+
+	AUXADCEnableSync(data->ref_source, data->sample_time, AUXADC_TRIGGER_MANUAL);
+	AUXADCGenManualTrigger();
+}
+
+static void adc_context_update_buffer_pointer(struct adc_context *ctx,
+					      bool repeat)
+{
+	struct adc_cc13xx_cc26xx_data *data =
+		CONTAINER_OF(ctx, struct adc_cc13xx_cc26xx_data, ctx);
+
+	if (repeat) {
+		data->buffer = data->repeat_buffer;
+	} else {
+		data->buffer++;
+	}
+}
+
+static int adc_cc13xx_cc26xx_init(const struct device *dev)
+{
+	struct adc_cc13xx_cc26xx_data *data = dev->data;
+	const struct adc_cc13xx_cc26xx_cfg *config = dev->config;
+
+	data->dev = dev;
+
+	/* clear any previous events */
+	AUXADCDisable();
+	HWREG(AUX_EVCTL_BASE + AUX_EVCTL_O_EVTOMCUFLAGSCLR) =
+		(AUX_EVCTL_EVTOMCUFLAGS_AUX_ADC_IRQ | AUX_EVCTL_EVTOMCUFLAGS_AUX_ADC_DONE);
+
+	config->irq_cfg_func();
+
+	adc_context_unlock_unconditionally(&data->ctx);
+	return 0;
+}
+
+static int adc_cc13xx_cc26xx_channel_setup(const struct device *dev,
+				    const struct adc_channel_cfg *channel_cfg)
+{
+	struct adc_cc13xx_cc26xx_data *data = dev->data;
+	const uint8_t ch = channel_cfg->channel_id;
+	uint16_t sample_time_us = 0;
+	uint8_t i;
+
+	if (ch > CHAN_COUNT) {
+		LOG_ERR("Channel 0x%X is not supported, max 0x%X", ch, CHAN_COUNT);
+		return -EINVAL;
+	}
+
+	switch (ADC_ACQ_TIME_UNIT(channel_cfg->acquisition_time)) {
+	case ADC_ACQ_TIME_TICKS:
+		data->sample_time = (uint16_t)ADC_ACQ_TIME_VALUE(channel_cfg->acquisition_time);
+		break;
+	case ADC_ACQ_TIME_MICROSECONDS:
+		sample_time_us = (uint16_t)ADC_ACQ_TIME_VALUE(channel_cfg->acquisition_time);
+		break;
+	case ADC_ACQ_TIME_NANOSECONDS:
+		sample_time_us = (uint16_t)(
+				ADC_ACQ_TIME_VALUE(channel_cfg->acquisition_time) * 1000);
+		break;
+	default:
+		data->sample_time = AUXADC_SAMPLE_TIME_170_US;
+		break;
+	}
+	if (sample_time_us) {
+		/* choose the nearest sample time configuration */
+		data->sample_time = adc_cc13xx_sample_times[0].reg_value;
+		for (i = 0; i < ARRAY_SIZE(adc_cc13xx_sample_times); i++) {
+			if (adc_cc13xx_sample_times[i].time_us > sample_time_us) {
+				break;
+			}
+			data->sample_time = adc_cc13xx_sample_times[i].reg_value;
+		}
```

Comment:
Shouldn't an error be returned when acquisition time > 10.9 ms is requested?

Reply:
I added a check if the iteration reached the for's end to return an error. Hope that's okay.